packages/flutter_tools/lib/src/macos/migrations/nsapplicationmain_deprecation_migration.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../xcode_project.dart';
+
+const String _appDelegateFileBefore = r'''
+@NSApplicationMain
+class AppDelegate''';
+
+const String _appDelegateFileAfter = r'''
+@main
+class AppDelegate''';
+
+/// Replace the deprecated `@NSApplicationMain` attribute with `@main`.
+///
+/// See:
+/// https://github.com/apple/swift-evolution/blob/main/proposals/0383-deprecate-uiapplicationmain-and-nsapplicationmain.md
+class NSApplicationMainDeprecationMigration extends ProjectMigrator {
```

Comment:
im wondering if it's possible to reuse the ios migration?

Reply:
They're very similar but subtly different - the affected file is different, the log messages are slightly different (due to differences in path), the attribute is different. The migration logic would need to be more a little more complicated to support both platforms. Personally I think this duplication results in simple code that's easier to maintain.